RandomizedTimeSeriesIT testRateGroupBySubset failing

### CI Link

https://gradle-enterprise.elastic.co/s/kuhbdwcnbasq4/tests/task/:x-pack:plugin:esql:internalClusterTest/details/org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT/testRateGroupBySubset?top-execution=1

### Repro line

./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT.testRateGroupBySubset" -Dtests.seed=6EA7BE49470FEE73 -Dtests.locale=ne-NP -Dtests.timezone=America/Manaus -Druntime.java=25	

### Does it reproduce?

Yes

### Applicable branches

9.3

### Failure history

_No response_

### Failure excerpt

```
java.lang.AssertionError: Failed. Agg: INCREASE | Failed windows: 1 windows(0.09606147934678194%):	
ROWS: 1041|Failed for row:	
[1, 1971.2245655979714, 1971.2245655979714, 1971.2245655979714, 1971.2245655979714, 1971.2245655979714, 2025-07-31T09:02:00.000Z, null, BsC, 62, hIMNGLHoJFCgSyE, null, uXStrAIUttPnGIK, ENJeCRItEpyKxQFc, cPMROM, null, null, null, K, null, null, 5107:de48::405b:62ca:ffff:ffff, RWmnwpVpse, null, 717]	
Wanted: RateStats[count=1, max=RateRange[lower=2598.5021090310906, upper=3175.947022149111], avg=RateRange[lower=2598.5021090310906, upper=3175.947022149111], min=RateRange[lower=2598.5021090310906, upper=3175.947022149111], sum=RateRange[lower=2598.5021090310906, upper=3175.947022149111]]	
Exception: 	
Expected: (a value less than or equal to <3175.947022149111> and not a value less than <2598.5021090310906>)	
     but: not a value less than <2598.5021090310906> was <1971.2245655979714>	
Row times and values:	
	TS:2025-07-31T09:02:32.843Z=811.0, 2025-07-31T09:02:00Z=916.0, 2025-07-31T09:02:37.631Z=758.0
```

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
@@ -418,6 +418,10 @@ public class RandomizedTimeSeriesIT extends AbstractEsqlIntegTestCase {
     ) {
         String timeseriesId = timeseries.getFirst().v1();
         var referenceTuple = isLowerBoundary ? timeseries.getFirst().v2() : timeseries.getLast().v2();
+        if (referenceTuple.v1().toEpochMilli() % (secondsInWindow * 1000L) == 0) {
+            // The reference tuple is already on the boundary.
+            return false;
+        }
         Tuple<Instant, Double> otherTuple = null;
         long otherTimestamp = 0;
         for (var doc : otherWindow) {

